Doc Update for Conflict Detection (#3396)
forcedotcom/salesforcedx-vscode: docs/_articles/en/user-guide/detect-conflicts.md

---
title: Detect Conflicts on Deploy
lang: en
---

> When you deploy source to your default org, the operation could detect conflicts on deploy to help you avoid accidentally overwriting others’ changes in the Org. The conflict detection feature is available for any Org that is not using source tracking, which is most non-scratch orgs such as sandboxes or dev orgs.

Reviewer: Because we're heavily pushing source tracking in "Developer" sandboxes for DevOps Center, let's think about how we can clarify. This is tricky.

Normally, I was taught to go positive vs. negative. What you can do vs. what you can't. But in this case it may be easier and simpler.

The conflict detection feature isn't available in orgs that use source tracking, such as scratch orgs and Developer sandboxes. 

Also, we made a choice at some point in SFDX doc pod to NOT capitalize "org".

Author: changed Org to org. Leaving the rest as it based on slack convo.

> When you deploy source to your default org, the operation could detect conflicts on deploy to help you avoid accidentally overwriting others’ changes in the Org. The conflict detection feature is available for any Org that is not using source tracking, which is most non-scratch orgs such as sandboxes or dev orgs.

Reviewer: ```suggestion
When you deploy source to your default org, the operation can detect conflicts on deploy to help you avoid accidentally overwriting others’ changes in the org. The conflict detection feature isn't available in orgs that use source tracking, such as scratch orgs and Developer sandboxes. (If you really want to drive it home). You can use this feature in non-source-tracked sandboxes and Developer Edition orgs. (and you could flip these two sentences if you like)
```

Reviewer: Would you ever use it when deploying to prod?

> A conflict occurs when the metadata file was last modified in the Org more recently than the last successful deployment or retrieval of this file to your local project.

Reviewer: ```suggestion
A conflict occurs when the metadata file in the org was last modified more recently than the last successful deployment or retrieval of this file to your local project.
```

If conflicts are detected in your project manifest or files, you can cancel the deploy operation and view the differences between your local files and default org.

> To view the files that caused conflicts, click the Org Differences icon () in the Activity Bar or the ‘Show Conflicts’ prompt from the dialog The Org Differences: Conflicts view opens in the Side Bar and displays the list of files with conflicts. To inspect conflicts, click a file to open the diff editor and compare the remote file (to the left) with the local file (on the right). To edit a file, hover over the file name and click the open file icon

Reviewer: ```suggestion
To view the files that caused conflicts, click the Org Differences icon () in the Activity Bar or the ***Show Conflicts*** prompt from the dialog. The Org Differences: Conflicts view opens in the side bar and displays the list of files with conflicts. To inspect conflicts, click a file to open the diff editor and compare the remote file (to the left) with the local file (on the right). To edit a file, hover over the file name and click the open file icon.
```

You can also view the resource files with conflicts and the CLI command to overwrite the conflicts in the Output panel.

If conflicts aren’t detected, the deploy operation is completed.

You must enable the conflict detection feature to use it:

1. Select **File** > **Preferences** > **Settings** (Windows or Linux) or **Code** > **Preferences** > **Settings** (macOS).

> 2. Under Salesforce Feature Previews, select Detect Conflicts At Sync.

Reviewer: ```suggestion
2. Under Salesforce Feature Previews, select **Detect Conflicts At Sync**.
```

> 2. Under Salesforce Feature Previews, select Detect Conflicts At Sync.

Reviewer: ```suggestion
2. Under Salesforce Feature Previews, select **Detect Conflicts At Sync**.
```

> 2. Under Salesforce Feature Previews, select Detect Conflicts At Sync.

Reviewer: ```suggestion
2. Under Salesforce Feature Previews, select **Detect Conflicts At Sync**.
```

> 2. Under Salesforce Feature Previews, select Detect Conflicts At Sync.

Reviewer: ```suggestion
2. Under Salesforce Feature Previews, select **Detect Conflicts At Sync**.
```

You can also enter conflict detection in the search box to find the feature and then enable it.

> We have enabled conflict detection for all Deploy commands executed from VS Code. All files that exist both in the org and in the local environment are checked for conflicts; files that don’t exist in both aren’t checked. If a conflict is detected on deploy, you can choose to view them or override them to continue the deploy operation:

Reviewer 2: ```suggestion
We have enabled conflict detection for all Deploy commands executed from VS Code. All files that exist both in the org and in the local environment are checked for conflicts; files that don’t exist in both aren’t checked. If conflicts are detected on deploy, you can choose to view them or override them to continue the deploy operation:
```
